Add an explanation of what error mypy produces when importing from a module with a top-line comment `# type: ignore`

**Documentation**

[Ignoring a whole file](https://mypy.readthedocs.io/en/stable/common_issues.html#ignoring-a-whole-file) explains the side effect of using `# type: ignore` at the top.

> A `# type: ignore` comment at the top of a module (before any statements, including imports or docstrings) has the effect of ignoring the entire contents of the module.

However, it doesn't show what kind of errors mypy produces when importing from such a module.
As shown in the following example, mypy provides an error styled of `error: Module "{{MODULE_NAME}}" has no attribute "{{ATTRIBUTE_NAME}}" [attr-defined]`.

Neither `has no attribute` nor `[attr-defined]` appears in [Ignoring a whole file](https://mypy.readthedocs.io/en/stable/common_issues.html#ignoring-a-whole-file) nor in the whole page of [Common issues and solutions](https://mypy.readthedocs.io/en/stable/common_issues.html).

It took me an hour to find this cause, and I believe it's worth mentioning in the documentation.

foo.py:
```python
from bar import bar_func, bar_value
```

bar.py:
```python
# type: ignore
bar_value = "bar"

def bar_func():
    pass
```

Executing mypy on foo.py:
```shell
> mypy foo.py
foo.py:1: error: Module "bar" has no attribute "bar_func"  [attr-defined]
foo.py:1: error: Module "bar" has no attribute "bar_value"  [attr-defined]
Found 2 errors in 1 file (checked 1 source file)
```

---

Environment:
```shell
> mypy --version
mypy 0.991 (compiled: yes)

> python3 --version
Python 3.11.1

> sw_vers
ProductName:	macOS
ProductVersion:	12.6.2
BuildVersion:	21G320

> sysctl -n machdep.cpu.brand_string
Apple M1 Pro
```

--- a/docs/source/common_issues.rst
+++ b/docs/source/common_issues.rst
@@ -188,6 +188,8 @@ Ignoring a whole file
 
 A ``# type: ignore`` comment at the top of a module (before any statements,
 including imports or docstrings) has the effect of ignoring the entire contents of the module.
+This behaviour can be surprising and result in
+"Module ... has no attribute ... [attr-defined]" errors.
 
 To only ignore errors, use a top-level ``# mypy: ignore-errors`` comment instead.
 To only ignore errors with a specific error code, use a top-level

